seqr/views/apis/individual_api.py

```diff
@@ -365,41 +365,56 @@ def _add_or_update_individuals_and_families(project, individual_records, user=No
     Return:
         2-tuple: updated_families, updated_individuals containing Django ORM models
     """
-    families = {}
+    updated_families = set()
     updated_individuals = set()
     parent_updates = []
-    for i, record in enumerate(individual_records):
-        # family id will be in different places in the json depending on whether it comes from a flat uploaded file or from the nested individual object
-        family_id = record.get(JsonConstants.FAMILY_ID_COLUMN) or record.get('family', {})['familyId']
-        family = families.get(family_id)
-        if family:
-            created = False
-        else:
-            family, created = Family.objects.get_or_create(project=project, family_id=family_id)
-
-        if created:
-            logger.info("Created family: %s", family)
 
-        # uploaded files do not have unique guid's so fall back to a combination of family and individualId
-        if record.get('individualGuid'):
-            individual_filters = {'guid': record['individualGuid']}
-        else:
-            individual_id = record.get(JsonConstants.PREVIOUS_INDIVIDUAL_ID_COLUMN) or record[JsonConstants.INDIVIDUAL_ID_COLUMN]
-            individual_filters = {'family': family, 'individual_id': individual_id}
+    family_ids = {_get_record_family_id(record) for record in individual_records}
+    families_by_id = {f.family_id: f for f in Family.objects.filter(project=project, family_id__in=family_ids)}
+
+    missing_family_ids = family_ids - set(families_by_id.keys())
+    for family_id in missing_family_ids:
+        family = Family.objects.create(project=project, family_id=family_id)
+        families_by_id[family_id] = family
+        updated_families.add(family)
+        logger.info('Created family: {}'.format(family))
+
+    individual_models = Individual.objects.filter(family__project=project).prefetch_related(
+        'family', 'mother', 'father')
+    has_individual_guid = any(record.get('individualGuid') for record in individual_records)
```

Comment:
It seems to me `any()` must be `all()`. Otherwise, it can't ensure all the records have an `individualGuid`.